src/components/ListProducts/ListProducts.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { FC } from "react";
+
+import Card from "@components/Card";
+import { ProductData } from "@type/index";
+import { useNavigate } from "react-router-dom";
+
+import styles from "./ListProducts.module.scss";
+
+interface ListProductsProps {
+  error: string;
+  data: ProductData[];
+}
+
+const ListProducts: FC<ListProductsProps> = ({ error, data }) => {
+  const navigate = useNavigate();
+
+  const handleCard = (id: number) => {
+    navigate(`/products/${id}`, { replace: true });
+  };
+
+  return (
+    <div className={styles.cards}>
+      {error ? (
+        <div>Ошибка</div>
+      ) : (
+        data &&
+        data.map((product: ProductData) => (
```

Comment:
```suggestion
        data?.map((product: ProductData) => (
```
а вообще это значение всегда является массивом, поэтому можно даже не проверять его, если вызвать `.map()` на пустом массиве, то ошибки не будет